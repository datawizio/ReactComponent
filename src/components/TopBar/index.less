@import "~antd/es/layout/style/index";
@import "../../styles/variables";

.top-bar {
  background-color: @layout-body-background;
  border-bottom: 1px solid @divider-border-color;
  display: flex;
  width: 100%;
  padding: 0 @padding-lg 0 0;
  align-items: center;
  .fill-element {
    flex: 1;
  }
  &.dw-dark {
    background-color: @bg-dark-color;
    border-bottom: none;
    box-shadow: 0 0 10px black;
    z-index: 100;
    .tutorial-link,
    .anticon-menu-unfold,
    .anticon-menu-fold {
      color: @icon-color-dark-theme;
      font-size: 18px;
    }
    .divider {
      border-right: 1px solid @divider-dark-border-color;
    }
  }

  .divider {
    height: 100%;
    border-right: 1px solid @divider-border-color;
    margin: 0 @margin-md;
  }
}
