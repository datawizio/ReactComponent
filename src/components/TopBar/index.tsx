import React from "react";

import { Layout as AntLayout } from "antd";

import "./index.less";
import clsx from "clsx";

export interface TopBarProps {
  theme?: "light" | "dark";
}

const TopBar: React.FC<TopBarProps> = ({ children, theme }) => {
  const className = clsx({
    "top-bar": true,
    "dw-dark": theme === "dark"
  });
  return <AntLayout.Header className={className}>{children}</AntLayout.Header>;
};

export default TopBar;
