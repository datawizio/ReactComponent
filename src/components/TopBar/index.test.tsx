import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import TopBar from "./index";

const mockProps = {
  theme: "ligth"
};

const setUp = (props?) => shallow(<TopBar {...props} />);

describe("TopBar component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });
  it("TopBar rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });
  it("TopBar ligth theme rendered correctly", () => {
    expect(component.hasClass("dw-dark")).toBeFalsy();
  });
  it("TopBar dark theme rendered correctly", () => {
    const wrapper = setUp({ theme: "dark" });
    expect(wrapper.hasClass("dw-dark")).toBeTruthy();
  });
});
