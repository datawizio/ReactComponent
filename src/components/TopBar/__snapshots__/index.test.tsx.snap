// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component TopBar rendered correctly 1`] = `
<Header
  className="top-bar"
/>
`;
